src/java/org/apache/cassandra/net/StartupClusterConnectivityChecker.java

```diff
@@ -170,12 +171,10 @@ public boolean execute(Set<InetAddressAndPort> peers, Function<InetAddressAndPor
         }
         else
         {
+
             // dc -> missing peer host addresses
             Map<String, List<String>> peersDown = acks.getMissingPeers().stream()
-                                                      .collect(groupingBy(peer -> {
-                                                                              String dc = peerToDatacenter.get(peer);
-                                                                              return dc != null ? dc : StringUtils.defaultString(getDatacenterSource.apply(peer), "unknown");
-                                                                          },
+                                                      .collect(groupingBy(peer -> Optional.ofNullable(peerToDatacenter.get(peer)).orElseGet(() -> StringUtils.defaultString(getDatacenterSource.apply(peer), "unknown")),
```

Comment:
@frankgh that is too verbose to my taste. Nothing wrong with what you had there before.

Reply:
I have reverted it back to the simple null checking

Reply:
@smiklosovic I have pushed one update that favors readability. I think we can go with this version of the code :)